pkg/imgutil/imgutil.go

```diff
@@ -102,6 +103,44 @@ func GetExistingImage(ctx context.Context, client *containerd.Client, snapshotte
 	return res, nil
 }
 
+func EnsureAllContent(ctx context.Context, client *containerd.Client, rawRef string, platform string, options types.GlobalCommandOptions) error {
+	named, err := distributionref.ParseDockerRef(rawRef)
+	if err != nil {
+		return err
+	}
+	refDomain := distributionref.Domain(named)
+
+	// Get a resolver
```

Comment:
We should put that away somewhere with the login code refactor. Resolver is repetitive and error prone...